### PR TITLE
Patch up the composer.json for the major branch 2

### PR DESCRIPTION
**Pull request by dnsl48**
https://github.com/silverstripe/cwp-recipe-kitchen-sink/issues/46

the build requires https://github.com/silverstripe/cwp-recipe-core/pull/13 to get merged first

### CODE REVIEW

**Review by robbieaverill — CHANGES_REQUESTED** [review]
CWP recipes target specific versions of everything you install. 2.4.x would be the next release, which is why the constraints are here. If there is a 2.4 branch created then you’d update the constraints in master to be the next minor for each of the lock stepped modules (CWP modules, recipes, and core modules and recipes). Other modules are probably less important but the core and cms constraints need to stay

File: .travis.yml [hunk 1]
```diff
@@ -1,14 +1,18 @@
...
 
 env:
   global:
-    - COMPOSER_ROOT_VERSION="2.4.x-dev" DB=MYSQL
+    - COMPOSER_ROOT_VERSION="2.x-dev" DB=MYSQL
 
 php:
   - 7.1
   - 7.2
+  - 7.3
```
Comment (robbieaverill) [inline .travis.yml:15]: This may not work yet, unless LDAP has been updated to support it

Comment (dnsl48) [reply .travis.yml:15]: just wanted to try :D
who knows...

**Comment by dnsl48** [comment]
need to rerun it when `cwp-recipe-core` is merged

**Comment by dnsl48** [comment]
> CWP recipes target specific versions of everything you install. 2.4.x would be the next release, which is why the constraints are here. If there is a 2.4 branch created then you’d update the constraints in master to be the next minor for each of the lock stepped modules (CWP modules, recipes, and core modules and recipes).

We'd like to try and keep those constraints within minor branches. That means we'd like to create minor branches and amend these constraints in there rather than updating masters.

File: .travis.yml (hunk 1 above)
Comment (dnsl48) [reply .travis.yml:15]: 7.3 is green now. Looks like we don't use LDAP in this recipe :-P

**Review by Cheddam — APPROVED** [review]
There have been further discussions around these concerns elsewhere, and we're ready to move forward with this.

**Pull request merged**